### PR TITLE
bugfix: move function getListener() to pkg/net/listener.go

### PR DESCRIPTION
**Pull request by YaoZengzeng**
Signed-off-by: YaoZengzeng <yaozengzeng@zju.edu.cn>

### CODE REVIEW

File: pkg/net/listener.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"syscall"
+
+	"github.com/alibaba/pouch/pkg/user"
```
Comment (allencloud) [inline pkg/net/listener.go]: No, I do not think a package in pkg still depends on a package in pouch.

File: pkg/net/listener.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+	"github.com/sirupsen/logrus"
+)
+
+func GetListener(addr string, tlsConfig *tls.Config) (net.Listener, error) {
+	addrParts := strings.SplitN(addr, "://", 2)
+	if len(addrParts) != 2 {
+		return nil, fmt.Errorf("invalid listening address: %s", addr)
```
Comment (allencloud) [inline pkg/net/listener.go]: I think in a common function, we could give more details about the error message. How about adding like this:
```
fmt.Errorf("invalid listening address %s: address must be [protocol]://[address]", addr)
```

**Comment by YaoZengzeng** [comment]
@fuweid PTAL.

File: pkg/netutils/listener_test.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		{
+			"invalidAddressTest",
+			args{"invalid address"},
+			fmt.Errorf("invalid listening address invalid address: must be in format [protocol]://[address]"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := GetListener(tt.args.addr, nil)
```
Comment (fuweid) [inline pkg/netutils/listener_test.go]: could we close it if the case pass?

**Pull request merged**